src/Hosting/TestHost/src/ClientHandler.cs

```diff
@@ -117,6 +117,13 @@ internal ClientHandler(PathString pathBase, ApplicationWrapper application)
                     {
                         req.Headers.Append(header.Key, string.Join(" ", header.Value));
                     }
+                    else if (string.Equals(header.Key, HeaderNames.ContentLength, StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (int.TryParse(header.Value.First(), out var contentLength))
```

Comment:
Could be a `long` I guess.
```suggestion
                        if (long.TryParse(header.Value.First(), out var contentLength))
```